bark/world/evaluation/rss_interface.cpp

```diff
@@ -0,0 +1,481 @@
+      ENUCoordinate(static_cast<double>(bg::get<1>(agent_center)));
+  matching_object.enuPosition.centerPoint.z = ENUCoordinate(0);
+  matching_object.enuPosition.heading = ::ad::map::point::createENUHeading(
+      static_cast<double>(agent_state(THETA_POSITION)));
+
+  // set object dimension/boundaries
+  matching_object.enuPosition.dimension.length = static_cast<double>(
+      Distance(agent_shape.front_dist_ + agent_shape.rear_dist_));
+  matching_object.enuPosition.dimension.width = static_cast<double>(
+      Distance(agent_shape.left_dist_ + agent_shape.right_dist_));
+  matching_object.enuPosition.dimension.height = Distance(2.);
+  matching_object.enuPosition.enuReferencePoint =
+      ::ad::map::access::getENUReferencePoint();
+
+  // perform map matching
+  ::ad::map::match::AdMapMatching map_matching;
+  matching_object.mapMatchedBoundingBox =
+      map_matching.getMapMatchedBoundingBox(matching_object.enuPosition);
+
+  return matching_object;
+}
+
+::ad::rss::map::RssObjectData RssInterface::GenerateObjectData(
+    const ::ad::rss::world::ObjectId& id,
+    const ::ad::rss::world::ObjectType& type,
+    const ::ad::map::match::Object& matchObject,
+    const ::ad::physics::Speed& speed,
+    const ::ad::physics::AngularVelocity& yawRate,
+    const ::ad::physics::Angle& steeringAngle,
+    const ::ad::rss::world::RssDynamics& rssDynamics) {
+  ::ad::rss::map::RssObjectData data = {
+      id, type, matchObject, speed, yawRate, steeringAngle, rssDynamics};
+  return data;
+}
+
+::ad::physics::AngularVelocity RssInterface::CaculateAgentAngularVelocity(
+    const models::dynamic::Trajectory& trajectory) {
+  int n = trajectory.rows();
+  if (n < 2) {
+    return 0.f;
+  } else {
+    ::ad::physics::AngularVelocity av = ::ad::physics::AngularVelocity(
+        trajectory(n - 1, THETA_POSITION) -
+        trajectory(0, THETA_POSITION) / trajectory(n - 1, TIME_POSITION) -
+        trajectory(0, TIME_POSITION));
+    return av;
+  }
+}
+
+FullRoute RssInterface::GenerateRoute(
+    const Point2d& agent_center,
+    const map::LaneCorridorPtr& agent_lane_corridor,
+    const ::ad::map::match::Object& match_object) {
+  // the center line of lane corridor is used as the target route
+  geometry::Line agent_lane_center_line = agent_lane_corridor->GetCenterLine();
+
+  float s_start = GetNearestS(agent_lane_center_line, agent_center);
+  float s_end = GetNearestS(
+      agent_lane_center_line,
+      agent_lane_center_line.obj_.at(agent_lane_center_line.obj_.size() - 1));
+
+  float step = (s_start < s_end) ? discretize_routing_step_ : -discretize_routing_step_;
+
+  std::vector<::ad::map::point::ENUPoint> routing_targets;
+  // discretize the line into points and store as routing targets
+  while (s_start <= s_end) {
+    geometry::Point2d traj_point = GetPointAtS(agent_lane_center_line, s_start);
+    routing_targets.push_back(::ad::map::point::createENUPoint(
+        bg::get<0>(traj_point), bg::get<1>(traj_point), 0));
```

Comment:
please use indents of 2 spaces everywhere

Reply:
It is already formatted with indent width of 2 spaces and Google style, the line 170 and 171 are too long and the vscode formatting plugin put it as two lines.